src/btree/bt_curprev.c

```diff
@@ -624,6 +626,23 @@ __wt_btcur_prev_prefix(WT_CURSOR_BTREE *cbt, WT_ITEM *prefix, bool truncating)
     for (newpage = false;; newpage = true, restart = false) {
         page = cbt->ref == NULL ? NULL : cbt->ref->page;
 
+        /*
+         * Determine if all records on the page have been deleted and all the tombstones are visible
+         * to our transaction. If so, we can avoid reading the records on the page and move to the
+         * next page. We base this decision on the aggregate stop point added to the page during the
+         * last reconciliation. We can skip this test if the page has been modified since it was
+         * reconciled or the underlying cursor is configured to ignore tombstones.
+         */
+        if (session->txn->isolation == WT_ISO_SNAPSHOT &&
+          !F_ISSET(&cbt->iface, WT_CURSTD_IGNORE_TOMBSTONE) && cbt->ref != NULL &&
+          __wt_ref_addr_copy(session, cbt->ref, &addr) == true && page != NULL &&
+          !__wt_page_is_modified(page) &&
+          __wt_txn_visible(session, addr.ta.newest_stop_txn, addr.ta.newest_stop_ts) &&
```

Comment:
Can we create a inline function for visibility checking of the aggregated time window similar to time window, like __wt_txn_tw_stop_visible or do we already have one?

Reply:
I believe there is no existing inline function that checks time aggregate data. The `__wt_txn_visible()` is already an inline function so I am assuming we will not gain much with making another enclosing inline function for time aggregate.

Reply:
If there is one, why we are not using it? We should standardize the way we do visibility checking.

Reply:
Sorry if I wasn't clear - there is no existing inline function that checks time aggregate window.

Reply:
got it